web/modules/custom/dpl_event/src/Plugin/rest/resource/v1/EventResource.php

```diff
@@ -114,6 +116,18 @@ public function patch(string $uuid, Request $request): Response {
     }
 
     $state = $request_data->getState();
+
+    if ($state) {
+      try {
+        $state_enum = EventState::from($state);
+        $state = $state_enum->value;
+        $event_instance->set('field_event_state', $state);
+      }
+      catch (\Error) {
```

Comment:
Consider why you choose to catch the more generic `\Error`.

You could instead catch `ValueError` which is what occurs if an enum cannot be extracted from a string. [`\Error`](https://www.php.net/manual/en/class.error.php) is the base class for all internal PHP errors and could thus be thrown in other situations as well. This would lead to `BadRequest` responses even though it is nothing that the client can do about the situation.

Also: I did not know you did not have to assign the exception being thrown to a variable. This is nice for situations where the exception is not relevant/used. Nice. I learned something new today.

Reply:
I chose Error, because if something goes wrong, it needs to inform the API consumer :)